sky/backends/backend_utils.py

```diff
@@ -867,6 +867,13 @@ def write_cluster_config(
     if to_provision.labels:
         labels.update(to_provision.labels)
 
+    # Prepare labels to be dumped into the yaml file - add quotes to the values
+    # to enforce str type for label values. Unquoted strings may get parsed as
+    # bool by yaml.safe_load (E.g., 'True' without quotes gets parsed as bool),
+    # which may cause issues with downstream code (e.g., Kubernetes python API
+    # does not accept bool values for labels).
+    labels = {k: f'\'{v}\'' for k, v in labels.items()}
```

Comment:
How about we fix it with the j2 logic?

https://github.com/skypilot-org/skypilot/blob/9011b1dc9c07950e8142de5fa6aa8a2077899cc2/sky/templates/aws-ray.yml.j2#L123